pkg/sql/catalog/funcdesc/func_desc.go

```diff
@@ -836,14 +836,36 @@ func IsOIDUserDefinedFunc(oid oid.Oid) bool {
 	return catid.IsOIDUserDefined(oid)
 }
 
-// CheckLeakProofVolatility returns an error when a function is defined as
+// ValidateLeakProofVolatility returns an error when a function is defined as
 // leakproof but not immutable. See more details in comments for volatility.V.
-func CheckLeakProofVolatility(fn catalog.FunctionDescriptor) error {
-	if fn.GetLeakProof() && fn.GetVolatility() != catpb.Function_IMMUTABLE {
-		return pgerror.Newf(
-			pgcode.InvalidFunctionDefinition,
+func ValidateLeakProofVolatility(
+	options tree.FunctionOptions, curLeakProof bool, curVolatility catpb.Function_Volatility,
+) error {
+	for _, option := range options {
+		switch t := option.(type) {
+		case tree.FunctionVolatility:
+			v, err := VolatilityToProto(t)
+			if err != nil {
+				panic(err)
```

Comment:
Why panic?

Reply:
huh, forgot to change to it to return error after moving it out of declarative schema changer.
Fixed.